Feat#1387: State of One Platform July blog update (#1064)
1-Platform/one-platform: packages/documentation-spa/blog/2021-08-09-state-of-one-platform-jul-2021.md

---
slug: state-of-one-platform-jul-2021
title: State of One Platform - Jul 2021
author: Ghanshyam Lohar
author_title: Senior Software Engineer @ Red Hat
author_url: https://github.com/ghanlohar
author_image_url: https://avatars.githubusercontent.com/u/5575651?s=400&v=4
tags: [redhat, blog, oneportal, oneplatform, react, angular, vue, jul, update]
---
### Meta
**Completed Sprints:** OP – Jul'21 Mid Sprint, OP – Jul'21 End Sprint

**Ongoing/Upcoming Sprint:** OP – Aug'21 Mid Sprint

**One Platform:** [one.redhat.com](https://one.redhat.com)

> **JIRA:** [https://projects.engineering.redhat.com/projects/ONEPLAT/issues](https://projects.engineering.redhat.com/projects/ONEPLAT/issues)

Reviewer: ```suggestion
**JIRA:** [https://issues.redhat.com/projects/ONEPLAT/issues](https://issues.redhat.com/projects/ONEPLAT/issues)
```

Reviewer 2: @riginoommen We don't need these links at all since even if issues.redhat.com is now external the JIRAs themselves are not accessible from outside

**GitHub:** [https://github.com/1-Platform/one-platform/issues](https://github.com/1-Platform/one-platform/issues)

### Introduction
This is a summary of the development goals achieved by the _One Platform team_ in the month of **July** 2021.

_This month, the strategy for the team was changed and the 4 weeks Sprint cycle was reduced to  2 Weeks cycle_.  As part of two Sprints in July’21, One Platform Team had the following high-level goals which have been achieved as part of **113** **Jira** Issues closed by the Team.

* **Application Development**:

>     * _[Home Page](https://one.redhat.com/) Enhancements_

Reviewer 2: ```suggestion
    * _[Home Page](https://one.redhat.com/) Enhancements_
```
Maybe useful in the future, but right now one.redhat.com is still internal

        * Collection of [Requirement](https://docs.google.com/spreadsheets/d/1Flk5hZRM7krzPInlTtfUGnJTN5ktfbyhonKAhEXxMiE/edit#gid=0)
        * Workflow creation to manage API Catalog with features like graphQL Playground/RESTful API Viewer/Swagger API Viewer
    * _Component Catalog _
        * Unified catalog listing component across Patternfly, OPC and Chapeaux
        * Technical conversation in progress. Requirement documents cover use cases for Phase I development.
    * _Innovation Hub_
        * UI development of Landing page.
        * UI implementation of Idea details page (title, description, total votes, option to vote etc.).
        * Create & update functionality fields for an Idea such as - Idea Summary and Description (user impact, value, audiences etc.).
        * _Production release tentatively on 31st Aug '21 date._
    * _Designs_
        * Re-defined the Designs for Search SPA, Notification SPA and Banner platform.
    * _Continuous Integration & Deployment (CI/CD) automation - _with pipeline for each in-built SPAs such as Notification SPA, User SPA, Feedback SPA etc.
    * Continuing SPAship operator onboarding for individual SPAs.
* **Platform Development**

>     * _[Developer Console ](https://stage.one.redhat.com/console/)_

Reviewer 2: ```suggestion
    * _[Developer Console ]_
```
Do we need an internal link?
But that also begs the question about the document (google doc) link, anyone outside redhat won't be able to access it.

        * App Settings page.
        * Integration with One Platform internal SPAs and the App Service.
        * App shell creation for Apps.
    * _[Lighthouse CI ](https://one.redhat.com/lighthouse)Release and GA_

>         * [Created a homepage for easy testing of apps based on URL input](https://projects.engineering.redhat.com/browse/ONEPLAT-1230).

Reviewer: ```suggestion
        * [Created a homepage for easy testing of apps based on URL input](https://issues.redhat.com/browse/ONEPLAT-1230).
```

>         * [Created a homepage for easy testing of apps based on URL input](https://projects.engineering.redhat.com/browse/ONEPLAT-1230).

Reviewer 2: ```suggestion
        * Created a homepage for easy testing of apps based on URL input.
```

>         * [Internal Property List Page](https://projects.engineering.redhat.com/browse/ONEPLAT-1231).

Reviewer: ```suggestion
        * [Internal Property List Page](https://issues.redhat.com/browse/ONEPLAT-1231).
```

>         * [Detailed analysis page for a SPA in a project](https://projects.engineering.redhat.com/browse/ONEPLAT-1232).

Reviewer: ```suggestion
        * [Detailed analysis page for a SPA in a project](https://issues.redhat.com/browse/ONEPLAT-1232).
```

>         * [Property-manager in service app](https://projects.engineering.redhat.com/browse/ONEPLAT-1240).

Reviewer: ```suggestion
        * [Property-manager in service app](https://issues.redhat.com/browse/ONEPLAT-1240).
```

        * Dashboard for showing the stats for Web properties.
        * Deployed Lighthouse backend to Openshift.

>     * _[GraphQL](https://api.qa.one.redhat.com/graphql)_

Reviewer: ```suggestion
    * _[GraphQL](https://api.qa.one.redhat.com/graphql)_
```

Why this url ? suggest to remove it as its

        * GraphQL federation support for microservices.
        * Deploy Graphql Portal in one platform.
* **Onboarding & Infrastructure update**

>     * [Attachment viewer](https://qa.one.redhat.com/attachment-viewer/) - Development is In-progress.

Reviewer 2: ```suggestion
    * Attachment viewer - Development is In-progress.
```

    * _Escalation watchlist & Pantheon Onboarding_ - Conversations In-progress.
    * _One Platform Externally accessible (outside VPN) _- Coordination In-progress with other teams such as - IT, Legal, InfoSec etc.
* **Non Native Apps**
    * **DSAL - Cost center tab release in DSAL Dashboard.**

### Up next in One Platform Store: ###

* Innovation hub GA release.
* App Development
* API Catalog Design and Backend.
* Component Catalog SPA.
* Lighthouse UI enhancements for better user experience.
* Developer Console adoption & enhancements.
* Reverse Proxy for upcoming experimental features.
* Notification framework and Banner configuration UI addition in Developer console.
* Explore GraphQL as a Managed Service.
* Whitelisting preprod environments in Akamai.
* One Platform readiness to be accessible outside VPN.

### What is One Platform and How does it benefit our Stakeholders? ###
Link to the Blog [here](https://source.redhat.com/groups/public/exd-digital-experience-platforms/exd_digital_experience_platforms_dxp_blog/part_i_why_one_platform_to_host_your_applications) (Please check out all parts - I,II & III)

### How to Report Issues with One Platform team : ###

>    * **JIRA:** [https://issues.redhat.com/projects/ONEPLAT/issues](https://issues.redhat.com/projects/ONEPLAT/issues)

Reviewer 2: ```suggestion
```
